obspy/core/utcdatetime.py

```diff
@@ -1204,6 +1157,16 @@ def __hash__(self):
         # explicitly flag it as unhashable
         return None
 
+    def __setattr__(self, key, value):
+        # raise a warning if overwriting previous ns (see #2072)
+        if self._initialized and not self._has_warned:
+            msg = ('Setting attributes on UTCDateTime instances will raise an'
+                   ' Exception in a future version of Obspy.')
+            warnings.warn(msg, ObsPyDeprecationWarning)
+            # only issue the warning once per object
+            self.__dict__['_has_warned'] = True
```

Comment:
Is it not Python's default behavior to only raise a warning message once? But I'm also sometimes a bit lost with how Python determines if a warning should be shown or not and it also behaves differently across versions.

Reply:
Yeah, I also think we should rely on Python managing how often this warning is shown.. it should only show it once by default anyway (it's keeping that magic per-module warning registry)?

Reply:
I thought the same thing but for some reason multiple warning were being issued from the same instance. I will look more into it.

Reply:
Ok, I can confirm that python 3.6 (on ubuntu 16)  raises 4 warnings for `test_change_precision_raises_warning` and two warnings for `test_compare_utc_different_precision_raises_warning` when line the `_has_warned` attr is not used. 

I have no idea why this is the case, but I am struggling to see a reason to be too concerned with leaving `_has_warned` in the code because it is only a temporary patch until `UTCDateTime` is immutable. If someone feels it is really important I will spend more time on it, but I would prefer to leave it alone.

Reply:
> I have no idea why this is the case, but I am struggling to see a reason to be too concerned with leaving _has_warned in the code because it is only a temporary patch until UTCDateTime is immutable. If someone feels it is really important I will spend more time on it, but I would prefer to leave it alone.

I agree. Let's leave it how you currently implemented it.

Reply:
```_has_warned``` mechanic is really ugly - but if its temporary please mark it in the code as temporary by adding an appropriate comment